src/viem/preparedTransactionSerialization.ts

```diff
@@ -0,0 +1,65 @@
+// Helper functions making prepared transactions serializable
+// so they can be used in redux actions (or even stores).
+import { TransactionRequestCIP42 } from 'node_modules/viem/_types/chains/celo/types'
+
+const bigIntProps = ['value', 'gas', 'maxFeePerGas', 'maxPriorityFeePerGas', 'gatewayFee'] as const
+
+type BigIntProps = (typeof bigIntProps)[number]
+
+type MapBigIntToString<T, K extends keyof T> = {
+  [P in keyof T]: P extends K ? string : T[P]
+}
+
+function mapBigIntsToStrings<T, K extends keyof T>(
+  obj: T,
+  bigIntKeys: readonly K[]
+): MapBigIntToString<T, K> {
+  const result = { ...obj } as MapBigIntToString<T, K>
+
+  for (const key of bigIntKeys) {
+    const value = obj[key]
+    if (typeof value === 'bigint') {
+      result[key] = value.toString() as any
+    }
+  }
+
+  return result
```

Comment:
```suggestion
  return bigIntKeys.reduce(
    (acc, key) => {
      const value = obj[key]
      return {
        ...acc,
        [key]: typeof value === 'bigint' ? value.toString() : value,
      }
    },
    { ...obj } as MapBigIntToString<T, K>
  )
```
I was looking for a way to avoid the "as any" type cast, looks like this would work

Reply:
I'll check, but does it work because TS is getting lost somehow? 😄

Reply:
Ok I did a couple of tests,
It does seem like TS is lost because of the spread and hence doesn't complain.
But it doesn't make the code safer.

See this: https://www.typescriptlang.org/play?#code/C4TwDgpgBAsghmAQgSwOYEkB2wAqB7AZWACdlNUAeHAGigGkoIAPYCTAEwGcoBrCEPADMoOAHxQAvFADeAKChQA2gAUoZXvyEiAugC4oq5qw7cGAfiicSZVFH04V22QF9ZswQFdMAY2DI8mFAAtggoGNic+ESk5JxUtAxGbFwaAsJiABTyUHgARgBW9tTZuWhYwHT8nPrEEHDsAQA2IPSKTgCU+vBIZdhR1uTx9OJyCrXAHsSBpeEVVQB0tewe3hBZCgoZcN7etHwg7ZIj2RtQ3gFWUABucI0e0FJ5+Yr7TqdjEBNTMifv8--bXa-U4vfh6KCgSBaG53B4SKQAchmZGACKgFkwEAA7rAEBlDvoYfdaAB6EkiAhQBoQTiYBHAM54IJgRpwMi0XIeBnIbm0+lQUioAAWwGBUFcp2cxVO0ig-3mT3FUDg3G6YXK-RilBow2y7RcQA